cmd/run.go

```diff
@@ -142,6 +148,31 @@ Run sidecar only:
 					os.Exit(1)
 				}
 
+				if appPort <= 0 {
+					// If app does not listen to port, we can check for Dapr's sidecar health before starting the app.
+					// Otherwise, it creates a deadlock.
+					sidecarUp := true
+					print.InfoStatusEvent(os.Stdout, "Checking if Dapr sidecar is listening on HTTP port %v", output.DaprHTTPPort)
+					err = utils.IsDaprListeningOnPort(output.DaprHTTPPort, time.Duration(runtimeWaitTimeoutInSeconds)*time.Second)
+					if err != nil {
+						sidecarUp = false
```

Comment:
sidecarUp will stay as false once it reaches here even if the second check to IsDaprListeningOnPort succeeds

Reply:
Correct. It means it is not listening to one of the ports.